UnitTests/TestMemProtector.cpp

```diff
@@ -58,3 +60,69 @@ TEST_CASE("Test setting page protections", "[MemProtector]") {
 	}
 	VirtualFree(page, 4 * 1024, MEM_RELEASE);
 }
+
+#else
```

Comment:
A big design decision for porting to version 2 of PolyHook was to not put significant logic behind if-defs but to instead properly model the challenges through the features of C++. Please check my previous partial port to linux for reference on how this should be modeled: https://github.com/stevemk14ebr/PolyHook_2_0/blob/042a34f54537b881e1c99fdd322421eb7352e35a/headers/MemoryAllocation/UnixImpl/UnixMemProtImp.hpp

Reply:
Basing the platform specific features on templates could cause problems that come with templates. Having this in one file is not great. It would be much better to have it in separate files and just include the right one based on ifdefs. What are your concerns with this solution?

Reply:
Templates aren't a problem. The protector is pretty simple, there should be a windows 'core' and a linux 'core' that get templated into a base object that uses their logic. This way the language verifies the interface is identical. I've dont the ifdef way it gets messy and convoluted.